doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,621 @@
+---
+title: "Introduction to igraph"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Introduction to igraph}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Tutorial
+
+The main goals of the igraph library is to provide a set of data types and functions for 1) pain-free implementation of graph algorithms, 2) fast handling of large graphs, with millions of vertices and edges, 3) allowing rapid prototyping via high level languages like R.
+
+## Creating a graph
+
+The simplest way to create a graph is the `Graph` constructor. To make an empty graph:
+
+```{r}
+library("igraph")
+
+gi <- make_empty_graph()
+```
+
+To make a graph with 10 nodes (numbered `1` to `10`) and two edges connecting nodes `1-2` and `1-5`:
+
+```{r}
+gi <- graph(edges = c(1,2, 1,5), n=10, directed = FALSE)
+```
+
+We can print the graph to get a summary of its nodes and edges:
+
+```{r echo = TRUE}
+print(gi)
+```
+
+This means: **U** ndirected graph with **10** vertices and **2** edges, with the exact edges listed out. If the graph has a [name] attribute, it is printed as well.
+
+Note
+
+`summary` is similar to `print` but does not list the edges, which is convenient for large graphs with millions of edges:
+
+```{r echo = TRUE}
+summary(gi)
+```
+
+## Adding/deleting vertices and edges
+
+Let's start from the empty graph again. To add vertices to an existing graph, use `add_vertices`:
+
+```{r}
+gi <- make_empty_graph()
+
+gi <- add_vertices(gi,3)
+```
+
+In igraph, vertices are always numbered up from one. The number of a vertex is called the *vertex ID*. A vertex might or might not have a name.
+
+Similarly, to add vertices use `add_edges`:
+
+```{r}
+gi <- add_edges(gi, edges = c(1,2, 1,3))
+```
+
+Edges are added by specifying the source and target vertex for each edge. This call added two edges, one connecting vertices `1` and `2`, and one connecting vertices `1` and `3`. Edges are also numbered up from one (the *edge ID*) and have an optional name.
+
+**Warning**
+
+Creating an empty graph and adding vertices and edges as shown here can be much slower than creating a graph with its vertices and edges as demonstrated earlier. If speed is of concern, you should especially avoid adding vertices and edges *one at a time*. If you need to do it anyway, you can use `add_vertex` and `add_edge`.
```

Comment:
We can try to rephrase to clarify:

"Repeatedly adding vertices/edges to a graph can be quite slow, especially if you are adding one edge/vertex at a time. If possible, create the graph with all its vertices and edges at once using `make_graph`. Otherwise, especially if speed is not important (small graphs, etc.) you can add a single vertex/edge at a time using `add_vertex` and `add_edge`."